app/controllers/shopping_lists_controller.rb

```diff
@@ -0,0 +1,27 @@
+class ShoppingListsController < ApplicationController
+  def index
+    @foods = Food.where(user_id: current_user.id)
+    @recipe_foods = RecipeFood.joins(recipe: :user).where(users: { id: current_user.id })
+    @food_quantities = @recipe_foods.group_by(&:food_id).transform_values { |foods| foods.sum(&:quantity) }
+    @greater_quantities = {}
+    total_price = 0
+    @food_quantities.each do |food_id, quantity|
+      food = Food.find_by(id: food_id)
+
+      next unless food && quantity > food.quantity
+
+      difference = quantity - food.quantity
+
+      @greater_quantities[food_id] = {
+        name: food.name,
+        quantity: difference,
+        measurement_unit: food.measurement_unit,
+        price: food.price * difference
+      }
+      total_price += @greater_quantities[food_id][:price]
+    end
+    @greater_quantities_array = @greater_quantities.to_a
+    @total_foods = @greater_quantities.size
+    @total_price = total_price
+  end
```

Comment:
- [ ] General shop list is not displaying as intended. Itemizing a Recipe by adding a number of ingredients are rendering correctly, however, when the `Generate Shopping list` but is clicked nothing is displayed on the `shopping_lists` screen. Let's ensure to improve upon this aspect.

**Add ingredient to a Recipe:**
![image](https://github.com/lRebornsl/recipe-app/assets/12745474/c1835d4c-dff0-42d9-a8b2-c119a944e470)

**Generate shopping list:**
![image](https://github.com/lRebornsl/recipe-app/assets/12745474/c38b991b-1a28-43ae-be82-d2cef39a0a63)

Reply:
@rloterh Thanks for the feedback! :+1:

Based on what you tell me, nothing is shown in the "shopping list" table because there is no food missing according to the list of foods that the user has. If you add more quantity than you have added to your grocery list, the "shopping list" will update showing you the missing quantity plus the missing value.

![image](https://github.com/lRebornsl/recipe-app/assets/103215273/ccccb9b4-cfa4-4239-88b8-1f868a2d2e48)

Reply:
@lRebornsl Kindly resubmit your project once again and it will be approved. I apologize for missing this step!